### PR TITLE
Feat/v5.1.0 remove profitbricks

### PR DESCRIPTION
**Pull request by gabisavu98**

### CODE REVIEW

File: ionoscloud/config.go [hunk 1]
```diff
@@ -16,22 +17,22 @@ type Config struct {
...
+		client = ionoscloud.NewAPIClient(ionoscloud.NewConfiguration(c.Username, c.Password, ""))
 	}
-	client.SetUserAgent(httpclient.TerraformUserAgent(terraformVersion))
+	client.GetConfig().UserAgent = httpclient.TerraformUserAgent(terraformVersion)
 
-	log.Printf("[DEBUG] Terraform client UA set to %s", client.GetUserAgent())
+	log.Printf("[DEBUG] Terraform client UA set to %s", client.GetConfig().UserAgent)
 
-	client.SetDepth(5)
+	client.GetConfig().AddDefaultQueryParam("depth", strconv.Itoa(5))
```
Comment (mflorin) [inline ionoscloud/config.go:32]: you don't need to `strconv.Itoa(5)` - you can just use the string "5" :)

File: ionoscloud/provider.go [hunk 2]
```diff
@@ -133,13 +132,13 @@ func providerConfigure(d *schema.ResourceData, terraformVersion string) (interfa
...
-		Retries:  d.Get("retries").(int),
-		Token:    token.(string),
-	}
+	//config := Config{
+	//	Username: username.(string),
+	//	Password: password.(string),
+	//	Endpoint: cleanedUrl,
+	//	Retries:  d.Get("retries").(int),
+	//	Token:    token.(string),
+	//}
```
Comment (mflorin) [inline ionoscloud/provider.go:141]: Same here, remove the commented code.

File: ionoscloud/provider.go [hunk 3]
```diff
@@ -12,13 +12,12 @@ import (
...
 	"github.com/hashicorp/terraform-plugin-sdk/terraform"
 	"github.com/ionos-cloud/sdk-go/v5"
-	"github.com/profitbricks/profitbricks-sdk-go/v5"
 )
 
-type SdkBundle struct {
-	LegacyClient *profitbricks.Client
-	Client       *ionoscloud.APIClient
-}
+//type SdkBundle struct {
```
[the remark this answers is not in the document]
Comment (gabisavu98) [reply ionoscloud/provider.go:17]: done

File: ionoscloud/provider.go (hunk 2 above)
Comment (gabisavu98) [reply ionoscloud/provider.go:141]: done

File: ionoscloud/config.go (hunk 1 above)
Comment (gabisavu98) [reply ionoscloud/config.go:32]: done

File: ionoscloud/utils.go [hunk 4]
```diff
@@ -79,12 +87,18 @@ func resourceK8sNodepoolImport(d *schema.ResourceData, meta interface{}) ([]*sch
...
+		defer cancel()
+	}
+
+	k8sNodepool, apiResponse, err := client.KubernetesApi.K8sNodepoolsFindById(ctx, parts[0], parts[1]).Execute()
 
 	if err != nil {
-		if apiError, ok := err.(profitbricks.ApiError); ok {
-			if apiError.HttpStatusCode() == 404 {
+		if _, ok := err.(ionoscloud.GenericOpenAPIError); ok {
+			if apiResponse.Response.StatusCode == 404 {
```
Comment (mflorin) [inline ionoscloud/utils.go:101]: apiResponse can be nil - you need to check that.

**Pull request merged**